Update provider documentation with deprecation dates (#163)
MetaMask/metamask-docs: docs/guide/provider-migration.md

# Provider Migration Guide

::: danger Deadline
All of these changes are shipping on **November 16, 2020**.

> If your web3 site relies on any of these features, it will break when they are removed unless you migrate.

Reviewer: Slight rewording, since at least one of these changes (the `eth_chainId` return value) is a change and not a removal, and _might_ not break anything for a site that used the feature.
```suggestion
If your web3 site relies on any of these features, it may break when these changes are made unless you migrate.
```
